README.md

```diff
@@ -1,57 +1,422 @@
+                    dictACRValues.Add(authContext.Id, authContext.DisplayName);
+                }
+
+                HttpContext.Session.Set<Dictionary<string, string>>(sessionKey, dictACRValues);
+            }
+        }
+        return dictACRValues;
+    }
+    ```
+
+    **CreateOrFetch** method checks if auth context exists then retrieve the list from graph by calling **ListAuthenticationContextClassReferencesAsync** method else call **CreateAuthContextViaGraph** to create the auth context.
+
+    ```csharp
+    public async Task<List<Beta.AuthenticationContextClassReference>> CreateOrFetch()
+    {
+        var lstPolicies = await _authContextClassReferencesOperations.ListAuthenticationContextClassReferencesAsync();
+    
+        if (lstPolicies?.Count > 0)
+        {
+            return lstPolicies;
+        }
+        else
+        {
+            await CreateAuthContextViaGraph();
+        }
+        return lstPolicies;
+    }
+    ```
+
+1. `AuthenticationContextClassReferencesOperations.cs` contains methods that call graph to perform various operations. In current sample we have used create and get methods. **CreateAuthenticationContextClassReferenceAsync** method creates the auth context: `Note: this class calls the /beta endpoint of Graph as the API was available only on the /beta endpoint at the time of this sample's publishing`
+
+    ```csharp
+   public async Task<Beta.AuthenticationContextClassReference> CreateAuthenticationContextClassReferenceAsync(string id, string displayName, string description, bool IsAvailable)
+    {
+        Beta.AuthenticationContextClassReference newACRObject = null;
+    
+        try
+        {
+            newACRObject = await _graphServiceClient.Identity.ConditionalAccess.AuthenticationContextClassReferences.Request().AddAsync(new Beta.AuthenticationContextClassReference
+            {
+                Id = id,
+                DisplayName = displayName,
+                Description = description,
+                IsAvailable = IsAvailable,
+                ODataType = null
+            });
+        }
+        catch (ServiceException e)
+        {
+            Console.WriteLine("We could not add a new ACR: " + e.Error.Message);
+            return null;
+        }
+    
+        return newACRObject;
+    }
+    ```
+
+    **ListAuthenticationContextClassReferencesAsync** method get the existing auth context values from graph.
+
+    ```csharp
+    public async Task<List<Beta.AuthenticationContextClassReference>> ListAuthenticationContextClassReferencesAsync()
+    {
+        List<Beta.AuthenticationContextClassReference> allAuthenticationContextClassReferences = new List<Beta.AuthenticationContextClassReference>();
+    
+        try
+        {
+            Beta.IConditionalAccessRootAuthenticationContextClassReferencesCollectionPage authenticationContextClassreferences = await _graphServiceClient.Identity.ConditionalAccess.AuthenticationContextClassReferences.Request().GetAsync();
```

Comment:
Suggestion - you could check IsAvailable flag for each of the contexts. I think the flag is corresponding to "Publish to apps" setting on portal, so when the setting is unchecked, we wouldn't want to see the specific context on the ToDo sample.

Reply:
Its actually more complicated than that. We have beenba sked to avoid that flag till its behavior finalized.
Will discuss more during the call